compiler/rustc_borrowck/src/renumber.rs

```diff
@@ -69,12 +69,10 @@ impl RegionCtxt {
     /// Used to determine the representative of a component in the strongly connected
     /// constraint graph
     pub(crate) fn preference_value(self) -> usize {
-        let _anon = Symbol::intern("anon");
-
         match self {
             RegionCtxt::Unknown => 1,
             RegionCtxt::Existential(None) => 2,
-            RegionCtxt::Existential(Some(_anon)) | RegionCtxt::Free(_anon) => 2,
+            RegionCtxt::Existential(Some(_)) | RegionCtxt::Free(_) => 2,
```

Comment:
Thanks for catching the mistake here, but using `_` here is not correct I believe. This should match for the `anon` symbol inside `RegionCtxt::Existential ` or `RegionCtxt::Free`.

Reply:
I just tried doing this and then I ran all the tests and they passed. So it doesn't seem to make much difference. In what case might this matter?

Reply:
I believe this code is only used for [`sccs_info`](https://github.com/rust-lang/rust/blob/8e1162f07e8b7d59f6a0e0a5576352d888defa78/compiler/rustc_borrowck/src/region_infer/mod.rs#L249), which is used to give information on region vars (for debugging purposes only currently).  I think we should prefer to use actual region names (if available, otherwise they are `anon`) over region information through `Location` or `TyContext` (though whether these are really more informative is debatable anyway I guess).